### ISSUE TITLE
Weight secondary photons spectrum to new injection spectral index

### ISSUE THREAD

**Issue by saikat8018** [CRPropa/CRPropa3#216]
Suppose, I run a simulation generating secondary photons from cosmic ray propagation with an initial injection spectral index a_i and propagate the photons below a crossover energy of 0.1*EeV with DINT. But DINT output provides only the photon flux in each energy bin. Is there a way by which I can find the photon spectrum for a different injection spectral index a_f. In the case of cosmic ray spectrum, we can give weights to the events while binning according to any spectral index we want. Is there any way to do that for DINT output?

**Comment by rafaelab** [comment]
This is currently not possible with DINT, as it is NOT the standard Monte Carlo simulation that you can just reweight. Therefore, you will need to simulate the UHECRs for each specific scenario, and then compute the photon fluxes with DINT.

**Comment by avvliet** [comment]
What is actually possible is to do the reweighting on the photon output file after you generate it with CRPropa and before you put it into DINT. So you do not have to rerun the CRPropa simulation, but you do have to rerun DINT.

**Comment by saikat8018** [comment]
Arjen, how can I implement this method, if DINT takes input only in `PhotonOutput1D()` format? By reweighting the photon output from CRPropa, I can generate the histogram of weighted events. But is it possible to pass this histogram to DINT?

**Comment by avvliet** [comment]
No, the input that you give to DINT has to be in the same format as the output of PhotonOutput1D(). So what you can do in the reweighting is, for each event in the output file, drawing a random number, see if that random number is smaller than the weight, and if it is write it to a new file with the same format as the output file. So you accept the particle when the weight is larger than the random number and reject it when the weight is smaller than the random number. You have to make sure, of course, that your weights are between 0 and 1 if your random numbers are between 0 and 1. This is rather cumbersome though, it might be easier to just rerun the simulation.

**Comment by cainaoliveira** [comment]
> No, the input that you give to DINT has to be in the same format as the output of PhotonOutput1D(). So what you can do in the reweighting is, for each event in the output file, drawing a random number, see if that random number is smaller than the weight, and if it is write it to a new file with the same format as the output file. So you accept the particle when the weight is larger than the random number and reject it when the weight is smaller than the random number. You have to make sure, of course, that your weights are between 0 and 1 if your random numbers are between 0 and 1. This is rather cumbersome though, it might be easier to just rerun the simulation.

Hi. If I want apply a exponential weight, like exp(-E/Ecut), this works too?

**Comment by lukasmerten** [comment]
@cainaoliveira Yes, the procedure @avvliet proposed will basically work with all weights that you can construct with the information available in your PhotonOutput1d()-like files.

**Comment by cainaoliveira** [comment]
> @cainaoliveira Yes, the procedure @avvliet proposed will basically work with all weights that you can construct with the information available in your PhotonOutput1d()-like files.

Thank you so much @lukasmerten!
